tests/ClientTest.php

```diff
@@ -805,4 +805,16 @@ public function testIdnWithRedirect()
         self::assertSame('https://xn--d1acpjx3f.xn--p1ai/images', (string) $request->getUri());
         self::assertSame('xn--d1acpjx3f.xn--p1ai', (string) $request->getHeaderLine('Host'));
     }
+
+    /**
+     * https://github.com/guzzle/guzzle/issues/2799
+     */
+    public function testGuzzleHeadRequestWithAcceptEncodingHeader(): void
+    {
+        $client = new Client(['headers' => ['accept-encoding' => 'gzip, deflate, br']]);
+
+        $response = $client->head('https://www.youtube.com/watch?v=kiHojsMTBeA');
```

Comment:
I believe this tests makes an actual network request. I am not sure whether this is a good idea. My 2 tests should detect the same bugs as this one, but go to the test server instead and check additional assertions.

Reply:
Oops =)

It is a good clip though.

Reply:
Fixed

Reply:
Hm.. no.. It does not show the bug now..hm..

Reply:
The mock response needs to have a `content-encoding: gzip` response header.